### PR TITLE
[StatsMigration] Add API in frontend to return stats report

### PR DESCRIPTION
**Pull request by justinlin-linkedin**
This PR adds a new API in frontend to return stats report to client.

### CODE REVIEW

File: ambry-frontend/src/main/java/com/github/ambry/frontend/GetStatsReportHandler.java [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+          } finally {
+            accountStatsStore.closeConnection();
+          }
+        }
+        if (snapshot == null) {
+          throw new RestServiceException("StatsReport not found for clusterName " + clusterName,
+              RestServiceErrorCode.NotFound);
+        }
+        try {
+          byte[] jsonBytes = mapper.writeValueAsBytes(snapshot);
```
Comment (jsjtzyy) [inline ambry-frontend/src/main/java/com/github/ambry/frontend/GetStatsReportHandler.java:151]: An alternative is to convert it to JSONObject and leverage `static ReadableStreamChannel serializeJsonToChannel(JSONObject jsonObject)` method in FrontendUtils.java

File: ambry-frontend/src/test/java/com/github/ambry/frontend/MockAccountStatsStoreFactory.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+  public MockAccountStatsStoreFactory(VerifiableProperties verifiableProperties, ClusterMapConfig clusterMapConfig,
+      StatsManagerConfig statsManagerConfig, MetricRegistry registry) {
+    accountStatsStore = new MockAccountStatsStore(clusterMapConfig.clusterMapClusterName);
+  }
+
+  @Override
+  public AccountStatsStore getAccountStatsStore() throws Exception {
+    return accountStatsStore;
+  }
```
Comment (jsjtzyy) [inline ambry-frontend/src/test/java/com/github/ambry/frontend/MockAccountStatsStoreFactory.java:38]: Can we use Mockito to replace this mock class?

**Review by jsjtzyy — COMMENTED** [review]
Mostly looks good to me. A few minor comments.

File: ambry-frontend/src/main/java/com/github/ambry/frontend/GetStatsReportHandler.java (hunk 1 above)
Comment (justinlin-linkedin) [reply ambry-frontend/src/main/java/com/github/ambry/frontend/GetStatsReportHandler.java:151]: that's unnecessary. ObjectMapper is more elegant.

File: ambry-frontend/src/test/java/com/github/ambry/frontend/MockAccountStatsStoreFactory.java (hunk 2 above)
Comment (justinlin-linkedin) [reply ambry-frontend/src/test/java/com/github/ambry/frontend/MockAccountStatsStoreFactory.java:38]: hmm, you are right, this is an unnecessary implementation.

**Review by lightningrob — APPROVED** [review]
lgtm

**Review by jsjtzyy — APPROVED** [review]

**Pull request merged**